backend/inspirehep/records/api/base.py

```diff
@@ -60,17 +65,41 @@ def validate(self):
         schema_validate(self)
 
     @classmethod
-    def get_uuid_from_pid_value(cls, pid_value, pid_type=None):
+    def get_uuid_from_pid_value(cls, pid_value, pid_type=None, original_record=False):
+        """Get uuid for provided PID value
+
+        Args:
+            pid_value(str): pid value to query
+            pid_type(str): pid_type to query
+            original_record(bool): if record was redirected this flag decides what this method should return
+              redirected record uuid (if False) or original record uuid (if True)
+
+        Returns: requested record uuid
+
+        """
         if not pid_type:
             pid_type = cls.pid_type
         pid = PersistentIdentifier.get(pid_type, pid_value)
+        while pid.is_redirected() and not original_record:
```

Comment:
do you need this `while` here? I think you already handle chains of redirects in the `get_redirect` method.

Reply:
Yes this should be an if. It was left from previous iteration of the code :/